src/vehicle_use.cpp

```diff
@@ -274,60 +274,89 @@ void vehicle::control_engines()
     int e_toggle = 0;
     bool dirty = false;
     //count active engines
-    int active_count = 0;
-    for( size_t e = 0; e < engines.size(); ++e ) {
-        if( is_part_on( engines[e] ) ) {
-            active_count++;
+    int active_mask = 0;
+    int fuel_count = 0;
+    int i = 0;
+    for( int e : engines ) {
+        if( is_part_on( e ) ) {
+            active_mask |= 1 << i++;
         }
-    }
+        fuel_count += part_info( e ).engine_fuel_opts().size();
+    }
+
+    const auto adjust_engine = [this]( int e_toggle ) {
+        int i = 0;
+        for( int e : engines ) {
+            for( const itype_id &fuel : part_info( e ).engine_fuel_opts() ) {
+                if( i == e_toggle ) {
+                    if( parts[ e ].ammo_current() == fuel ) {
+                        toggle_specific_part( e, !is_part_on( e ) );
+                    } else {
+                        parts[ e ].ammo_set( fuel, 0 );
+                    }
+                    return;
+                }
+                i += 1;
+            }
+        }
+        return;
```

Comment:
Why? There is nothing after this in the function anyway.

Reply:
It used to return a value when I was debugging; thanks for the catch.